src/idfxt/JSONDataInterface.h

```diff
@@ -0,0 +1,102 @@
+// copyright 2014, Shannon Mackey <mackey@BUILDlab.net>
+#ifndef JSONDATAINTERFACE_H
+#define JSONDATAINTERFACE_H
+
+//for JSONDataObject and JSONDataInterface
+#include <string>
+#include <list>
+#include <map>
+#include <memory>
+
+//for old IDF text import only
+#include <algorithm>
+#include <cctype>
+#include <cstring>
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <string>
+#include <vector>
+#include <chrono>
+#include <random>
+
+const std::string ACCEPTED_VERSION = "8.2";
+//end old IDF text import only
+
+extern "C"
+{
+#include "../../third_party/cJSON/cJSON.h"
+}
+
+namespace idfx {
+
+
+class JSONDataObject
+{
+public:
+    JSONDataObject(const std::string &json_content);
+    ~JSONDataObject();
+
+    std::map<std::string, std::string> getProperties();
+    std::string getPropertyValue(std::string);
+//    void setPropertyValue(std::string, std::string);
+//    void setPropertyValue(std::string, int);
+//    void setPropertyValue(std::string, double);
+
+    std::list<std::shared_ptr<JSONDataObject> > getExtensions();
+
+    
+    std::string print();
+    //friend access private constructor to avoid conversion to/from strings internally
+//    friend std::list<JSONDataObject> JSONDataInterface::getModelObjects(std::string);
+
+
+private:
+    cJSON *data_j;
+
+    void c_getChildren(std::list< std::shared_ptr< idfx::JSONDataObject > > &return_object_list, cJSON *subitem);
+//    JSONDataObject(cJSON &cjson); //to hide cJSON pointer
+};
+
+
+//////////////////////////////////////////////////////////
+
+class JSONDataInterface
+{
+public:
+    JSONDataInterface(const std::string &json_schema);
+    ~JSONDataInterface();
+
+    std::list<std::shared_ptr<JSONDataObject> > getModelObjects(std::string object_type = "");
+
+    std::list<JSONDataObject> getModelObjectDataByIndex(
```

Comment:
@refaqtor I am a little confused by this.  If I understand correctly, JSONDataInterface is somewhat equivalent to IdfFile and JSONDataObject is somewhat equivalent to IdfObject, is that right?  If so, shouldn't a uuid be enough to find a particular object?  What does the int index do?  Also, why is a list returned?  If the number of objects is either 0-1 then maybe returning a null pointer would be preferred to an empty list?  @kbenne can weigh in on that.